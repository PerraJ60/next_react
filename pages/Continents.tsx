import Layout from 'src/components/Layout';
import { ApolloClient, InMemoryCache, gql } from '@apollo/client';
import styles from 'styles/Home.module.css';

export default function Continents({ continents }) {
  console.log('continents', continents);
  return (
    <Layout>
      <div className={styles.container}>
        <main className={styles.main}>
          <h1 className={styles.title}>All Continents</h1>

          <p className={styles.description}>Continents</p>

          <div className={styles.grid}>
            {continents.map((continent) => {
              return (
                <a key={continent.code} className={styles.card}>
                  <h3>{continent.name}</h3>
                  <h5>Code: {continent.code}</h5>
                </a>
              );
            })}
          </div>
        </main>
      </div>
    </Layout>
  );
}

export async function getStaticProps() {
  const client = new ApolloClient({
    uri: 'https://countries.trevorblades.com/',
    cache: new InMemoryCache(),
  });

  const { data } = await client.query({
    query: gql`
      query GetContinents {
        continents {
          name
          code
        }
      }
    `,
  });

  return {
    props: {
      continents: data.continents,
    },
  };
}
