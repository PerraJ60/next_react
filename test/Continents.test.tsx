import TestRenderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import '@testing-library/jest-dom/extend-expect';
import Continents, { continentsQuery } from '../pages/Continents';

afterEach(cleanup);

const mocks = [
  {
    request: {
      query: continentsQuery,
      variables: {},
    },
    result: {
      data: [],
    },
  },
];

it('renders without error', () => {
  const component = TestRenderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <Continents continents={mocks} />
    </MockedProvider>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
