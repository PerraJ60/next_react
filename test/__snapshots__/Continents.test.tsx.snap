// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without error 1`] = `
<div>
  <nav>
    <ul
      className="header"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/Products"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Products
        </a>
      </li>
      <li>
        <a
          href="/Continents"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Continents
        </a>
      </li>
      <li>
        <a
          href="/Spacex"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Space travels
        </a>
      </li>
      <li>
        <a
          href="/About"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          CV
        </a>
      </li>
      <li>
        <a
          href="/Contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="container"
  >
    <main
      className="main"
    >
      <h1
        className="title"
      >
        All Continents
      </h1>
      <p
        className="description"
      >
        Continents
      </p>
      <div
        className="grid"
      >
        <a
          className="card"
        >
          <h3 />
          <h5>
            Code: 
          </h5>
        </a>
      </div>
    </main>
  </div>
</div>
`;
